go/backend/archive/ldb/archive.go

```diff
@@ -239,18 +242,18 @@ func (a *Archive) Exists(block uint64, account common.Address) (exists bool, err
 	return exists, err
 }
 
-func (a *Archive) GetBalance(block uint64, account common.Address) (balance common.Balance, err error) {
+func (a *Archive) GetBalance(block uint64, account common.Address) (balance amount.Amount, err error) {
 	var key accountBlockKey
 	key.set(backend.BalanceArchiveKey, account, block)
 	keyRange := key.getRange()
 	it := a.db.NewIterator(&keyRange, nil)
 	defer it.Release()
 
 	if it.Next() {
-		copy(balance[:], it.Value())
+		balance = amount.NewFromBytes(it.Value()[:]...)
 		return balance, nil
 	}
-	return common.Balance{}, it.Error()
+	return amount.New(), it.Error()
```

Comment:
should not we have `amount.Zero()` ?

Reply:
Once it's all merged, I'll create a new PR with `Zero` factory method and all `New()` instances replaced.